packages/holocron/src/holocronModule.jsx

```diff
@@ -96,33 +96,36 @@ export default function holocronModule({
   mergeProps,
   options = {},
 } = {}) {
+  let that;
   return function wrapWithHolocron(WrappedComponent) {
     class HolocronModuleWrapper extends React.Component {
       constructor(props) {
         super(props);
         this.state = {
           loadCount: 0,
           status: 'loading',
+          currentProps: props,
         };
         this.mounted = false;
+        that = this;
       }
 
       componentDidMount() {
         this.mounted = true;
         this.initiateLoad(0, this.props);
       }
 
-      // ignoring to support deprecated componentWillReceiveProps.
-      // This needs to be removed to support React17
-      // eslint-disable-next-line camelcase
-      UNSAFE_componentWillReceiveProps(nextProps) {
-        const { loadCount } = this.state;
-        const { props } = this;
-        if (shouldModuleReload && shouldModuleReload(props, nextProps)) {
+      static getDerivedStateFromProps(nextProps, { currentProps, loadCount }) {
+        if (shouldModuleReload && shouldModuleReload(currentProps, nextProps)) {
           const newLoadCount = loadCount + 1;
-          this.setState({ status: 'loading', loadCount: newLoadCount });
-          this.initiateLoad(newLoadCount, nextProps);
+          that.initiateLoad(newLoadCount, nextProps);
+          return {
+            status: 'loading',
+            loadCount: newLoadCount,
+            currentProps: nextProps,
```

Comment:
`currentProps` will only be updated if a reload is triggered. Future comparisons will be out of date.